src/log/Log.cc

```diff
@@ -312,13 +312,13 @@ void Log::_flush(EntryVector& t, bool crash)
         syslog(LOG_USER|LOG_INFO, "%s", pos);
       }
 
-      if (do_stderr) {
-        std::cerr << m_log_stderr_prefix << std::string_view(pos, used) << std::endl;
-      }
-
       /* now add newline */
       pos[used++] = '\n';
 
+      if (do_stderr) {
+        std::clog << m_log_stderr_prefix << std::string_view(pos, used) << std::flush;
```

Comment:
`std::clog` is not mandated to be buffered. Best to just use a direct `::writev` instead.

Reply:
`writev` is not in C++ standard, so we may have portability issues. Besides, `man stderr` said:
> Note that mixing use of FILEs and raw file descriptors can produce unexpected results and should generally be avoided.

And implementing a `safe_writev` would introduce some extra complexity.

How about `fmt::print(stderr, ...)`?

Reply:
I just tested with `strace` on Ubuntu 20.04. Though not mentioned in the document, `fmt::print` just issues one `write` system call for one log line. `std::clog` is indeed not buffered, and issues 2 system calls.

And I did the benchmark, `fmt::print` is as performant as `writev`

Reply:
`writev` is part of the unix standard and available on all unices systems. Windows has `WSASend` [1].

I don't think you need to worry about handling `EINTR` for "safe io". This is just for logging and EINTR is extraordinarily rare in my experience.

[1] https://docs.microsoft.com/en-us/windows/win32/api/winsock2/nf-winsock2-wsasend?redirectedfrom=MSDN

Reply:
can we just go with `fmt::print()` ? as it is more straightforward.